## Terraform Provider Tests - FAILED ✗

**Commit:** `4d6787f`
**Duration:** 1m 45s
**Results:** 121 passed, 1 failed, 5 skipped

### Failed Tests

| Test | Error |
|------|-------|
| `TestTeam` | Test failed |

<details>
<summary>Full test output (click to expand)</summary>

```
=== RUN   TestAccVantageAccessGrant_basic
--- PASS: TestAccVantageAccessGrant_basic (8.94s)
=== RUN   TestAccAnomalyNotification_basic
--- PASS: TestAccAnomalyNotification_basic (1.01s)
=== RUN   TestAccBillingProfile_basic
--- PASS: TestAccBillingProfile_basic (0.74s)
=== RUN   TestAccBillingProfile_withNestedAttributes
--- PASS: TestAccBillingProfile_withNestedAttributes (0.66s)
=== RUN   TestAccBillingProfile_withCompleteNestedAttributes
--- PASS: TestAccBillingProfile_withCompleteNestedAttributes (0.66s)
=== RUN   TestAccBillingProfile_withBankingAttributes
--- PASS: TestAccBillingProfile_withBankingAttributes (0.65s)
=== RUN   TestAccBillingProfile_withBothNestedAttributes
--- PASS: TestAccBillingProfile_withBothNestedAttributes (0.38s)
=== RUN   TestAccBillingProfilesDataSource_basic
--- PASS: TestAccBillingProfilesDataSource_basic (0.34s)
=== RUN   TestAccBillingProfilesDataSource_withResource
--- PASS: TestAccBillingProfilesDataSource_withResource (0.40s)
=== RUN   TestAccBillingProfilesDataSource_multipleAdjustmentItems
    billing_profiles_data_source_test.go:85: Skipping test: MSP invoicing required (MANAGED_ACCOUNT_DOMAIN not set)
--- SKIP: TestAccBillingProfilesDataSource_multipleAdjustmentItems (0.00s)
=== RUN   TestAccBillingRule_basic
--- PASS: TestAccBillingRule_basic (3.56s)
=== RUN   TestAccVantageBudget_basic
--- PASS: TestAccVantageBudget_basic (1.04s)
=== RUN   TestAccVantageBudget_withPeriodsUpdate
--- PASS: TestAccVantageBudget_withPeriodsUpdate (0.81s)
=== RUN   TestAccVantageBudget_withEmptyPeriods
--- PASS: TestAccVantageBudget_withEmptyPeriods (0.80s)
=== RUN   TestAccVantageBudget_multipleChildBudgets
--- PASS: TestAccVantageBudget_multipleChildBudgets (1.13s)
=== RUN   TestCostReportAttachmentLabelForAPI
--- PASS: TestCostReportAttachmentLabelForAPI (0.00s)
=== RUN   TestBusinessMetricOmittedAttachmentLabelPayload
--- PASS: TestBusinessMetricOmittedAttachmentLabelPayload (0.00s)
=== RUN   TestAssignCostReportTokens_OrderPreservation
--- PASS: TestAssignCostReportTokens_OrderPreservation (0.00s)
=== RUN   TestAssignCostReportTokens_NullLabelFilter
--- PASS: TestAssignCostReportTokens_NullLabelFilter (0.00s)
=== RUN   TestAccBusinessMetric_basic
--- PASS: TestAccBusinessMetric_basic (1.79s)
=== RUN   TestAccBusinessMetric_valuesKnownInUpdatePlan
--- PASS: TestAccBusinessMetric_valuesKnownInUpdatePlan (0.65s)
=== RUN   TestAccBusinessMetric_cloudwatch
    business_metric_resource_test.go:516: Skipping until we have support for Integrations/AccessCredentials
--- SKIP: TestAccBusinessMetric_cloudwatch (0.00s)
=== RUN   TestAccBusinessMetric_datadog
    business_metric_resource_test.go:596: Skipping until we have support for Integrations/AccessCredentials
--- SKIP: TestAccBusinessMetric_datadog (0.00s)
=== RUN   TestAccBusinessMetric_costReportTokensOrder
--- PASS: TestAccBusinessMetric_costReportTokensOrder (0.95s)
=== RUN   TestAccBusinessMetric_costReportMetadataLabelAndCalculationType
--- PASS: TestAccBusinessMetric_costReportMetadataLabelAndCalculationType (0.95s)
=== RUN   TestAccBusinessMetric_withValuesAndEmptyLabelFilter
--- PASS: TestAccBusinessMetric_withValuesAndEmptyLabelFilter (0.78s)
=== RUN   TestAccBusinessMetric_costReportTokensWithReferences
--- PASS: TestAccBusinessMetric_costReportTokensWithReferences (1.03s)
=== RUN   TestAccBusinessMetric_forecastedValues
--- PASS: TestAccBusinessMetric_forecastedValues (1.12s)
=== RUN   TestAccBusinessMetric_withValuesAndForecastedValues
--- PASS: TestAccBusinessMetric_withValuesAndForecastedValues (0.69s)
=== RUN   TestBusinessMetricSnowflakeFieldsPayload
=== RUN   TestBusinessMetricSnowflakeFieldsPayload/toCreate
=== RUN   TestBusinessMetricSnowflakeFieldsPayload/toUpdate
--- PASS: TestBusinessMetricSnowflakeFieldsPayload (0.00s)
    --- PASS: TestBusinessMetricSnowflakeFieldsPayload/toCreate (0.00s)
    --- PASS: TestBusinessMetricSnowflakeFieldsPayload/toUpdate (0.00s)
=== RUN   TestBusinessMetricLabelFiltersPayload
--- PASS: TestBusinessMetricLabelFiltersPayload (0.00s)
=== RUN   TestLabelFiltersToAPIEmptyMap
=== RUN   TestLabelFiltersToAPIEmptyMap/null
=== RUN   TestLabelFiltersToAPIEmptyMap/unknown
=== RUN   TestLabelFiltersToAPIEmptyMap/empty
--- PASS: TestLabelFiltersToAPIEmptyMap (0.00s)
    --- PASS: TestLabelFiltersToAPIEmptyMap/null (0.00s)
    --- PASS: TestLabelFiltersToAPIEmptyMap/unknown (0.00s)
    --- PASS: TestLabelFiltersToAPIEmptyMap/empty (0.00s)
=== RUN   TestCanvasDataObject
=== RUN   TestCanvasDataObject/nil_data
=== RUN   TestCanvasDataObject/table_and_error
--- PASS: TestCanvasDataObject (0.00s)
    --- PASS: TestCanvasDataObject/nil_data (0.00s)
    --- PASS: TestCanvasDataObject/table_and_error (0.00s)
=== RUN   TestCanvasTableJSON
=== RUN   TestCanvasTableJSON/nil_table
--- PASS: TestCanvasTableJSON (0.00s)
    --- PASS: TestCanvasTableJSON/nil_table (0.00s)
=== RUN   TestAccVantageCanvas_basic
--- PASS: TestAccVantageCanvas_basic (0.85s)
=== RUN   TestAccVantageCanvas_updatePrompt
--- PASS: TestAccVantageCanvas_updatePrompt (0.79s)
=== RUN   TestAccVantageCostAlert_withMinimumThreshold
--- PASS: TestAccVantageCostAlert_withMinimumThreshold (1.16s)
=== RUN   TestCostAlert
--- PASS: TestCostAlert (1.96s)
=== RUN   TestAccCostReport_basic
--- PASS: TestAccCostReport_basic (2.06s)
=== RUN   TestAccCostReport_grouping
--- PASS: TestAccCostReport_grouping (0.65s)
=== RUN   TestAccCostReport_settings
--- PASS: TestAccCostReport_settings (0.82s)
=== RUN   TestAccCostReport_chartSettings
--- PASS: TestAccCostReport_chartSettings (0.79s)
=== RUN   TestAccCustomProviderCostsUploadResource_basic
--- PASS: TestAccCustomProviderCostsUploadResource_basic (0.67s)
=== RUN   TestAccCustomProviderCostsUploadResource_badHeaders
--- PASS: TestAccCustomProviderCostsUploadResource_badHeaders (0.23s)
=== RUN   TestAccCustomProviderCostsUploadResource_autoTransform
--- PASS: TestAccCustomProviderCostsUploadResource_autoTransform (0.59s)
=== RUN   TestAccCustomProviderCostsUploadResource_customFilename
--- PASS: TestAccCustomProviderCostsUploadResource_customFilename (0.43s)
=== RUN   TestAccCustomProviderResource_basic
--- PASS: TestAccCustomProviderResource_basic (0.66s)
=== RUN   TestAccCustomProviderResource_withDescription
--- PASS: TestAccCustomProviderResource_withDescription (0.57s)
=== RUN   TestAccCustomProviderResource_withWorkspaces
--- PASS: TestAccCustomProviderResource_withWorkspaces (0.89s)
=== RUN   TestAccDashboard_basic
--- PASS: TestAccDashboard_basic (1.33s)
=== RUN   TestAccDashboard_withCostReportWidget
--- PASS: TestAccDashboard_withCostReportWidget (0.42s)
=== RUN   TestAccDashboard_hasDateInterval
--- PASS: TestAccDashboard_hasDateInterval (0.95s)
=== RUN   TestAccDashboard_dateInterval
--- PASS: TestAccDashboard_dateInterval (1.40s)
=== RUN   TestAccDashboard_nullDateIntervalNoDrift
--- PASS: TestAccDashboard_nullDateIntervalNoDrift (1.23s)
=== RUN   TestAccFinancialCommitmentReportsDataSource_basic
--- PASS: TestAccFinancialCommitmentReportsDataSource_basic (0.64s)
=== RUN   TestFinancialCommitmentReport
--- PASS: TestFinancialCommitmentReport (0.91s)
=== RUN   TestFinancialCommitmentReport_withEmptyGroupings
--- PASS: TestFinancialCommitmentReport_withEmptyGroupings (0.67s)
=== RUN   TestAccVantageFolderDataSource_basic
--- PASS: TestAccVantageFolderDataSource_basic (0.41s)
=== RUN   TestAccVantageFolderDataSource_withWorkspaceFilter
--- PASS: TestAccVantageFolderDataSource_withWorkspaceFilter (0.42s)
=== RUN   TestFetchAllFolders_singlePage
--- PASS: TestFetchAllFolders_singlePage (0.00s)
=== RUN   TestFetchAllFolders_multiplePages
--- PASS: TestFetchAllFolders_multiplePages (0.00s)
=== RUN   TestFetchAllFolders_emptyResult
--- PASS: TestFetchAllFolders_emptyResult (0.00s)
=== RUN   TestFetchAllFolders_stopsAtNilNext
--- PASS: TestFetchAllFolders_stopsAtNilNext (0.00s)
=== RUN   TestFetchAllFolders_apiError
--- PASS: TestFetchAllFolders_apiError (0.00s)
=== RUN   TestAccVantageFolder_basic
--- PASS: TestAccVantageFolder_basic (0.64s)
=== RUN   TestAccVantageFolder_withSavedFilterTokens
--- PASS: TestAccVantageFolder_withSavedFilterTokens (0.97s)
=== RUN   TestAccVantageFolder_preservesSavedFilterTokensWhenOmitted
--- PASS: TestAccVantageFolder_preservesSavedFilterTokensWhenOmitted (0.87s)
=== RUN   TestAccIntegrationByNameDataSource_basic
--- PASS: TestAccIntegrationByNameDataSource_basic (0.40s)
=== RUN   TestAccIntegrationByNameDataSource_withProviderFilter
--- PASS: TestAccIntegrationByNameDataSource_withProviderFilter (0.44s)
=== RUN   TestAccIntegrationsDataSource_basic
--- PASS: TestAccIntegrationsDataSource_basic (0.43s)
=== RUN   TestAccIntegrationsDataSource_withProviderFilter
--- PASS: TestAccIntegrationsDataSource_withProviderFilter (0.48s)
=== RUN   TestFetchAllIntegrations_singlePage
--- PASS: TestFetchAllIntegrations_singlePage (0.00s)
=== RUN   TestFetchAllIntegrations_multiplePages
--- PASS: TestFetchAllIntegrations_multiplePages (0.00s)
=== RUN   TestFetchAllIntegrations_emptyResult
--- PASS: TestFetchAllIntegrations_emptyResult (0.00s)
=== RUN   TestFetchAllIntegrations_withProviderFilter
--- PASS: TestFetchAllIntegrations_withProviderFilter (0.00s)
=== RUN   TestFetchAllIntegrations_stopsAtNilNext
--- PASS: TestFetchAllIntegrations_stopsAtNilNext (0.00s)
=== RUN   TestFetchAllIntegrations_apiError
--- PASS: TestFetchAllIntegrations_apiError (0.00s)
=== RUN   TestAccInvoice_basic
    invoice_resource_test.go:12: Invoice tests require a managed account token - should be enabled when managed accounts are available
--- SKIP: TestAccInvoice_basic (0.00s)
=== RUN   TestAccInvoicesDataSource_basic
--- PASS: TestAccInvoicesDataSource_basic (0.29s)
=== RUN   TestKubernetesReport
--- PASS: TestKubernetesReport (0.93s)
=== RUN   TestKubernetesReport_withEmptyGroupings
--- PASS: TestKubernetesReport_withEmptyGroupings (0.66s)
=== RUN   TestAccKubernetesEfficiencyReportsDataSource_basic
--- PASS: TestAccKubernetesEfficiencyReportsDataSource_basic (0.59s)
=== RUN   TestAccManagedAccount_basic
--- PASS: TestAccManagedAccount_basic (1.31s)
=== RUN   TestAccManagedAccount_withEmailDomain
--- PASS: TestAccManagedAccount_withEmailDomain (0.74s)
=== RUN   TestAccVantageManagedAccountsDataSource_basic
--- PASS: TestAccVantageManagedAccountsDataSource_basic (0.47s)
=== RUN   TestNetworkFlowReport
--- PASS: TestNetworkFlowReport (0.91s)
=== RUN   TestNetworkFlowReport_withEmptyGroupings
--- PASS: TestNetworkFlowReport_withEmptyGroupings (0.64s)
=== RUN   TestAccNetworkFlowReportsDataSource_basic
--- PASS: TestAccNetworkFlowReportsDataSource_basic (0.60s)
=== RUN   TestPageFromURL
=== RUN   TestPageFromURL/standard_next-page_URL
=== RUN   TestPageFromURL/page_10
=== RUN   TestPageFromURL/missing_page_parameter
=== RUN   TestPageFromURL/non-integer_page_value
=== RUN   TestPageFromURL/invalid_URL
--- PASS: TestPageFromURL (0.00s)
    --- PASS: TestPageFromURL/standard_next-page_URL (0.00s)
    --- PASS: TestPageFromURL/page_10 (0.00s)
    --- PASS: TestPageFromURL/missing_page_parameter (0.00s)
    --- PASS: TestPageFromURL/non-integer_page_value (0.00s)
    --- PASS: TestPageFromURL/invalid_URL (0.00s)
=== RUN   TestAccVantageRecommendationView_basic
--- PASS: TestAccVantageRecommendationView_basic (0.75s)
=== RUN   TestAccVantageRecommendationView_withFilters
--- PASS: TestAccVantageRecommendationView_withFilters (0.82s)
=== RUN   TestAccVantageRecommendationView_withRegions
--- PASS: TestAccVantageRecommendationView_withRegions (0.64s)
=== RUN   TestAccVantageRecommendationView_withDateFilters
--- PASS: TestAccVantageRecommendationView_withDateFilters (0.64s)
=== RUN   TestAccVantageRecommendationView_withTags
--- PASS: TestAccVantageRecommendationView_withTags (0.61s)
=== RUN   TestAccVantageRecommendationView_allFilters
--- PASS: TestAccVantageRecommendationView_allFilters (0.38s)
=== RUN   TestAccVantageReportForecast_basic
--- PASS: TestAccVantageReportForecast_basic (1.19s)
=== RUN   TestAccVantageReportForecast_clearBusinessMetric
--- PASS: TestAccVantageReportForecast_clearBusinessMetric (1.07s)
=== RUN   TestAccVantageReportForecastsDataSource_basic
--- PASS: TestAccVantageReportForecastsDataSource_basic (0.61s)
=== RUN   TestAccVantageReportNotification_basic
--- PASS: TestAccVantageReportNotification_basic (0.56s)
=== RUN   TestAccResourceReportColumnsDataSource
--- PASS: TestAccResourceReportColumnsDataSource (0.40s)
=== RUN   TestAccResourceReportColumnsDataSourceInvalidResourceType
--- PASS: TestAccResourceReportColumnsDataSourceInvalidResourceType (0.10s)
=== RUN   TestAccResourceReport
--- PASS: TestAccResourceReport (0.89s)
=== RUN   TestAccResourceReportWithCustomColumns
--- PASS: TestAccResourceReportWithCustomColumns (0.91s)
=== RUN   TestAccResourceReportWithCustomColumnsMultiResourceTypeError
--- PASS: TestAccResourceReportWithCustomColumnsMultiResourceTypeError (0.23s)
=== RUN   TestAccVantageResourceReportsDataSource_basic
    resource_reports_data_source_test.go:14: Skipping while we resolve AtoI error
--- SKIP: TestAccVantageResourceReportsDataSource_basic (0.00s)
=== RUN   TestAccVantageSavedFilter_basic
--- PASS: TestAccVantageSavedFilter_basic (0.64s)
=== RUN   TestAccVantageScenarioModel_basic
--- PASS: TestAccVantageScenarioModel_basic (0.82s)
=== RUN   TestScenarioModelWorkspaceTokenPreservesComputedState
--- PASS: TestScenarioModelWorkspaceTokenPreservesComputedState (0.00s)
=== RUN   TestAccVantageScenarioModelsDataSource_basic
--- PASS: TestAccVantageScenarioModelsDataSource_basic (0.49s)
=== RUN   TestAccSegment_basic
--- PASS: TestAccSegment_basic (0.94s)
=== RUN   TestTeam
    team_resource_test.go:16: Step 6/8 error running import: ImportStateVerify attributes not equivalent. Difference is shown below. The - symbol indicates attributes missing after import.
        
          map[string]string{
        - 	"role": "editor",
          }
--- FAIL: TestTeam (1.58s)
=== RUN   TestTeamDefaultDashboardToken
--- PASS: TestTeamDefaultDashboardToken (1.03s)
=== RUN   TestVirtualTagConfig_ApplyPayload_NilNestedListsAreKnownEmpty
--- PASS: TestVirtualTagConfig_ApplyPayload_NilNestedListsAreKnownEmpty (0.00s)
=== RUN   TestVirtualTagConfig_ApplyPayload_PopulatedNestedLists
--- PASS: TestVirtualTagConfig_ApplyPayload_PopulatedNestedLists (0.00s)
=== RUN   TestAccVantageVirtualTagConfig_basic
--- PASS: TestAccVantageVirtualTagConfig_basic (2.48s)
=== RUN   TestAccVantageVirtualTagConfig_withDateRanges
--- PASS: TestAccVantageVirtualTagConfig_withDateRanges (1.94s)
=== RUN   TestAccVantageVirtualTagConfig_withLabelTransforms
--- PASS: TestAccVantageVirtualTagConfig_withLabelTransforms (1.16s)
=== RUN   TestAccVantageVirtualTagConfig_optionalListsConsistent
--- PASS: TestAccVantageVirtualTagConfig_optionalListsConsistent (0.54s)
=== RUN   TestAccVantageWorkspaceDataSource_basic
--- PASS: TestAccVantageWorkspaceDataSource_basic (0.40s)
=== RUN   TestFetchAllWorkspaces_singlePage
--- PASS: TestFetchAllWorkspaces_singlePage (0.00s)
=== RUN   TestFetchAllWorkspaces_multiplePages
--- PASS: TestFetchAllWorkspaces_multiplePages (0.00s)
=== RUN   TestFetchAllWorkspaces_emptyResult
--- PASS: TestFetchAllWorkspaces_emptyResult (0.00s)
=== RUN   TestFetchAllWorkspaces_stopsAtNilNext
--- PASS: TestFetchAllWorkspaces_stopsAtNilNext (0.00s)
=== RUN   TestFetchAllWorkspaces_apiError
--- PASS: TestFetchAllWorkspaces_apiError (0.00s)
=== RUN   TestAccVantageWorkspace
--- PASS: TestAccVantageWorkspace (0.79s)
=== RUN   TestAccVantageWorkspace_invalidCurrencyConversion
--- PASS: TestAccVantageWorkspace_invalidCurrencyConversion (0.08s)
=== RUN   TestAccVantageWorkspacesDataSource_basic
--- PASS: TestAccVantageWorkspacesDataSource_basic (0.29s)
FAIL
FAIL	github.com/vantage-sh/terraform-provider-vantage/vantage	84.975s
FAIL
```

</details>

---
To re-run tests, a maintainer can comment `/test` on this PR.

--- a/vantage/team_resource.go
+++ b/vantage/team_resource.go
@@ -224,6 +224,12 @@ func (r TeamResource) Read(ctx context.Context, req resource.ReadRequest, resp *
 	}
 	state.WorkspaceTokens = workspaceTokensValue
 
+	// Role is not returned by the API, so preserve the create/update default
+	// when initializing state during import.
+	if state.Role.IsNull() || state.Role.IsUnknown() {
+		state.Role = types.StringValue("editor")
+	}
+
 	resp.Diagnostics.Append(resp.State.Set(ctx, &state)...)
 }
 

